OpenRA.Mods.Common/Pathfinder/PathCacheStorage.cs

```diff
@@ -9,12 +9,77 @@
  */
 #endregion
 
+using System;
 using System.Collections.Generic;
 using System.Linq;
 
 namespace OpenRA.Mods.Common.Pathfinder
 {
-	public class PathCacheStorage : ICacheStorage<List<CPos>>
+	public enum PathCacheQueryType : byte
+	{
+		UnitPath,
+		UnitPathToRange
+	}
+
+	public readonly struct PathCacheKey
+	{
+		readonly PathCacheQueryType queryType;
+		readonly uint actorID;
+		readonly int source;
+		readonly int target;
+		readonly int targetY;
+		readonly int hash;
+
+		public PathCacheKey(PathCacheQueryType queryType, uint actorID, CPos source, CPos target)
```

Comment:
This ctor only makes sense with `PathCacheQueryType.UnitPath` and the one below with `PathCacheQueryType.UnitPathToRange`, so we can remove that arg and have it implicit based on which of the two is used (or can it be removed completely?).